structuredheadings/plugin.js

```diff
@@ -28,26 +38,56 @@
     }
   };
 
+  var clearNumbering = function (editor, element) {
+    if (element.hasClass(editor.config.autonumberBaseClass)) {
+      element.removeClass(editor.config.autonumberBaseClass);
+    }
+  };
+
+  var setNumbering = function (editor, element) {
+    if (!element.hasClass(editor.config.autonumberBaseClass)) {
+      element.addClass(editor.config.autonumberBaseClass);
+    }
+  };
+
+  var clearLevel = function (editor, element) {
+    for (var key in editor.config.autonumberLevelClasses) {
+      element.removeClass(editor.config.autonumberLevelClasses[key]);
+    }
+  };
+
+  var setLevel = function (editor, element) {
+    var index = editor.config.numberedElements.indexOf(element.getName());
+    clearLevel(editor, element);
+    element.addClass(editor.config.autonumberLevelClasses[index]);
+  };
+
   var clearStyles = function (editor, element) {
     for (var styleName in editor.config.autonumberStyles) {
       var style = editor.config.autonumberStyles[styleName];
-      for (var className in style) {
-        element.removeClass(style[className]);
+      if (typeof style === "object") {
+        for (var className in style) {
+          element.removeClass(style[className]);
+        }
+      } else {
+        element.removeClass(style);
```

Comment:
The only way for style to be null is if someone edited the config and supplied blank styles to overwrite the default values. If that were the case, style being null would be the least of your problems, the editor likely wouldn't load.

In this case, style is a CSS class